platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -180,10 +179,15 @@ public Response sendLogoutRequest(@QueryParam("EncryptedNameIdTime") String encr
           return buildLogoutResponse(msg);
         }
 
+        SecurityToken idpSecToken = getIdpSecurityToken();
```

Comment:
Why are local variables used to hold the token? Why not call `getIdpSecurityToken` when the token is needed?

Reply:
I used a local variable because we want to check if it is null first, but then we use that variable again later. This leads to avoid calling `getIdpSecurityToken` again.

Reply:
What are the negative consequences of calling `getSecurityToken` again?

Reply:
Calling `getSecurityToken` calls `getTokenHolder` which calls `getOrCreateSession` which has the potential to try and create a new session. This could be a problem in this case when trying to handle the correct session and assertions.

Reply:
I personally prefer having a local variable for the token here, since we use the token twice, and don't need the unnecessary (and probably little) overhead of retrieving the token twice.